code/game/machinery/hivebot_swarm_core.dm

```diff
@@ -0,0 +1,124 @@
+/obj/machinery/hivebot_swarm_core //A hulking machine armed with several weapons. Creates hivebots and defends itself with a buzzsaw and mounted weapons.
+	name = "hivebot swarm core"
+	desc = "An incredibly dangerous machine with random weapons haphazardly attached with spot welds. Its main body seems impervious to damage, but the head is vulnerable."
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "hivebot_swarm_core"
+	pixel_x = -32
+	pixel_y = -16
+	use_power = FALSE
+	obj_integrity = 250
+	max_integrity = 250
+	anchored = TRUE
+	density = TRUE
+	var/recharge_period = 0 //The number of ticks the core is waiting for and cannot do anything
+	var/atom/movable/threat_to_swarm //The core's "target" for its attacks
+	var/hivebot_limit = 40 //Stop making hivebots if there are this many in a 7x7 square around us
+
+/obj/machinery/hivebot_swarm_core/Destroy()
+	visible_message("<span class='warning'>[src] falls apart, its red eyes sputtering out.</span>")
+	playsound(src, 'sound/magic/clockwork/anima_fragment_death.ogg', 50, 0)
+	new/obj/structure/fluff/hivebot_swarm_core(get_turf(src))
+	return ..()
+
+/obj/machinery/hivebot_swarm_core/examine(mob/user)
+	..()
+	if(recharge_period)
+		to_chat(user, "<span class='warning'>It seems to be recharging.</span>")
+
+/obj/machinery/hivebot_swarm_core/process()
+	if(recharge_period)
+		recharge_period--
+		return
+	acquire_target()
+	if(!threat_to_swarm)
+		make_hivebot()
+	else
+		defend_the_swarm()
+
+/obj/machinery/hivebot_swarm_core/proc/make_hivebot() //Only called if we don't find a threat
+	var/hivebots = 0
+	for(var/mob/living/simple_animal/hostile/hivebot/H in range(7, src))
+		hivebots++
+	if(hivebots >= hivebot_limit)
+		return
+	var/turf/T = get_step(src, SOUTHWEST)
+	var/mob/living/simple_animal/hostile/hivebot/H = new(T)
+	H.color = rgb(0, 255, 0)
+	animate(H, color = initial(H.color), time = 10)
+
+/obj/machinery/hivebot_swarm_core/proc/acquire_target() //If we find a mob in our view, it becomes the threat. Robots don't count because we're too stupid to recognize them.
+	if(threat_to_swarm)
+		return
+	var/list/potential_targets = list()
+	for(var/mob/living/carbon/C in view(7, src))
+		potential_targets += C
+	for(var/mob/living/silicon/S in view(7, src))
+		potential_targets += S
+	for(var/obj/mecha/M in view(7, src))
+		potential_targets += M
+	for(var/mob/living/L in potential_targets)
+		if(L.stat)
+			potential_targets -= L
+		else
+			for(var/faction in L.faction)
+				if(faction == "hivebot")
```

Comment:
? you should just use the faction var, it's an atom var, so you can set it on the machine iirc

then just do if(src.faction in L.faction)